frontend/packages/console-app/src/components/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,279 @@
+      <dt>Name</dt>
+      <dd>
+        <ResourceIcon kind="PersistentVolumeClaim" />
+        {pvc}
+      </dd>
+      <dt>Namespace</dt>
+      <dd>
+        <ResourceIcon kind="Namespace" />
+        {namespace}
+      </dd>
+      <dt>Status</dt>
+      <dd>
+        <Status status={status} />
+      </dd>
+      <dt>Storage Class</dt>
+      <dd>
+        <ResourceIcon kind="StorageClass" />
+        {storageClass}
+      </dd>
+      <dt>Requested Capacity</dt>
+      <dd>{requestedCapacity}</dd>
+      <dt>Access Mode</dt>
+      <dd>{accessModes}</dd>
+      <dt>Volume Mode</dt>
+      <dd>{volumeMode}</dd>
+    </dl>
+  );
+};
+
+const CreateSnapshotForm: React.FC<SnapshotResourceProps> = (props) => {
+  const { resourceName, namespace, kindObj } = props;
+
+  const [pvcName, setPVCName] = React.useState(resourceName || '');
+  const [pvcObj, setPVCObj] = React.useState(null);
+  const [snapshotName, setSnapshotName] = React.useState<string>(`${pvcName}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState<string>();
+  const [error, setError] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [pvcStatus, setPVCStatus] = React.useState<string>(null);
+  const [pvcProvisioner, setPVCProvisioner] = React.useState<string>(null);
+
+  React.useEffect(() => {
+    const setValue = async () => {
+      try {
+        const pvc = await fetchK8s<K8sResourceKind>(PersistentVolumeClaimModel, pvcName, namespace);
+        const sc = await fetchK8s<StorageClassResourceKind>(
+          StorageClassModel,
+          pvc?.spec?.storageClassName,
+        );
+        setPVCObj(pvc);
+        setPVCStatus(pvc?.metadata?.deletionTimestamp ? 'Terminating' : pvc?.status?.phase);
+        setPVCProvisioner(sc?.provisioner);
+      } catch (err) {
+        setError(err);
+      }
+    };
+
+    setValue();
+  }, [pvcName, namespace]);
+
+  const handleSnapshotName: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    setSnapshotName(event.currentTarget.value);
+  };
+
+  const create = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    setInProgress(true);
+    const snapshotTemplate: K8sResourceKind = {
+      apiVersion: `${VolumeSnapshotModel.apiGroup}/${VolumeSnapshotModel.apiVersion}`,
+      kind: VolumeSnapshotModel.kind,
+      metadata: {
+        name: snapshotName,
+        namespace,
+      },
+      spec: {
+        volumeSnapshotClassName: snapshotClassName,
+        source: {
+          persistentVolumeClaimName: pvcName,
+        },
+      },
+    };
+    k8sCreate(VolumeSnapshotModel, snapshotTemplate)
+      .then((resource) => {
+        setInProgress(false);
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      })
+      .catch((err) => {
+        setError(err.message || 'Could not create volume snapshot');
+        setInProgress(false);
```

Comment:
You can create a finally block for `setInProgress(false)`

`
.finally(() => setInProgress(false);
`